lec17-rmarkdown.Rmd

```diff
@@ -1,4 +1,306 @@
+to the actual variable or code containing that value rather than manually writing
+it out. Here is an example:
+
+There are `r ''``r nrow(df)` samples in this experiment.
+
+
+### Images and Figures
+To include an image, use the following syntax:
+```
+[caption for my image](path/to/image.png)
+```
+If you would like the caption you wrote to be included underneath your image,
+put the following in your YAML metadata:
+
+```
+---
+title: Document
+output:
+    html_document:
+        fig_caption: yes
+
+```
+As you've learned in doing your assignments, both the code and output of your
+`r` code chunk such as figures will show up in your output document. However,
+using code chunk options `echo` and `eval`, you can suppress output of the code
+underlying a graph, and only show the resulting plot from your code or vice versa.
+In this case, you probably want the former, and it would look like this:
+
+`r ''````{r, eval=TRUE, echo=FALSE}
+library(ggplot2)
+qplot(mpg, wt, data=mtcars)
+```
+
+You can also set the figure height and width too using the `fig.height` and
+`fig.width` options.
+
+`r ''````{r, fig.width=7, fig.height=7, eval=TRUE, echo=FALSE}
+library(ggplot2)
+qplot(mpg, wt, data=mtcars)
+```
+
+### In-Line Citations & Bibliography
+
+Pandoc can automatically generate citations and a bibliography in a number of
+styles. In order to use this feature, you will need to specify a bibliography
+file using the bibliography metadata field in a YAML metadata section.
+For example:
+
+```
+---
+title: "Sample Document"
+output: html_document
+bibliography: bibliography.bib
+---
+```
+
+Many bibliography formats are accepted (see the [R Markdown guides](http://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html)),
+such as a .bib file which many citation managers can generate for you and will
+hold all the references you need for your document (e.g. [EndNote](https://www.reed.edu/cis/help/LaTeX/EndNote.html))
+
+Every entry in your bibliography file should have a shorthand key id which
+when preceded by '@' allows you to reference the citation in-line. They usually
+go within square brackets and are separated by semicolons.
+
+```
+One Citation:
+Some fact [@Smith2014]
+
+Multiple Citations:
+Statement [@Smith2014; @Logan1997].
+```
+
+To make a bibliography, you may also want to specify a citation style guide to
+format your bibliography (in the form of a .csl file). The official repository
+for recognized citation styles is available here: https://github.com/citation-style-language/styles
+
+#### Exercise
+Take some time to find the style guide for your field and to download that file.
+
+Once you have the correct .csl file, you can specify it in your YAML metadata:
+
+```
+---
+title: "Sample Document"
+output: html_document
+bibliography: bibliography.bib
+csl: nature.csl
+---
+```
+
+### Footnotes
+
+```
+This is what a footnote looks like.[^1] Here is another.[^2]
+
+[^1]: My first footnote.
+[^2]: My second footnote.
```

Comment:
Also include that one of the strengths about Pandoc footnotes is that you can reference them by name and don't have to renumber them as the text is reordered (http://pandoc.org/MANUAL.html#footnotes).

> The identifiers in footnote references may not contain spaces, tabs, or newlines. These identifiers are used only to correlate the footnote reference with the note itself; in the output, footnotes will be numbered sequentially.

The same goes for figures actually, at least with fignos extension to pandoc, note sure about R Markdown.